### ISSUE TITLE
Include Point geometries in folium module to plot nodes interactively 

### ISSUE THREAD

**Issue by PyMap** [gboeing/osmnx#802]
Hi @gboeing!
I'm working on a `streamlit` project where users can get bbox coordinates by clicking on a web map. The main goal is to let them  create their own street networks and get some relevant stats to study the mobility in the target area.  

The [folium](https://github.com/gboeing/osmnx/blob/main/osmnx/folium.py) module looks great to add some interactivity to the analysis but I faced that it only supports LineStrings geometries. And since I do not want to only describe the graph with edges but also with nodes, I was looking for some extra methods to allow plotting them based on their connectivity attributes.

Specifically, I needed to render Point geometries on a Leaflet web map and get them colored by some attribute (betweenness centrality, degree centrality, node degree, etc) with the `cmap` bar legend/reference. Something like this:

![image](https://user-images.githubusercontent.com/44501803/154883528-55a7b508-e8d5-4967-adac-b6a7226cab94.png)

Below I share you the code and some examples:

1. [build a Point geoms gdf and add node colors based on some attribute](https://github.com/PyMap/ssyt/blob/main/ssyt/mobilidad.py#L56-L89)
2. [create a circle marker map based on color attributes](https://github.com/PyMap/ssyt/blob/main/ssyt/mobilidad.py#L91-L157)
3. [some examples can be found here](https://github.com/PyMap/ssyt/blob/main/ssyt/accesibility.ipynb)
 
If you find this could be an interesting feature to add in the folium.py module we can have some further discussions about the CircleMarker map implementation. 

Thanks!

**Comment by gboeing** [comment]
This could be a nice feature to add to the folium module for richer exploratory analysis. Would you like to sketch out some ideas for implementation?

**Comment by PyMap** [comment]
I've been thinking that nodes can be plotted in two different ways:

**1. regardless the [plot_graph_folium](https://github.com/gboeing/osmnx/blob/main/osmnx/folium.py#L15-L74) method.**

This option contemplates some nested methods with default parameters to only plot nodes over a leaflet layer. Basically what I referenced above. This implies to:

1.1. Calculate the centrality metric to be plot.

I first thought to use the `ox.basic_stats(G)` or `ox.extended_stats(G)` methods to get node attributes, but since they will be deprecated I think the plotting methods should directly receive a G containing the fields to be plotted. This implies to calculate them outside the plotting methods. Something like this:

```
# Calculates degree centrality with networkX
dc = nx.degree_centrality(G)
nx.set_node_attributes(G, values=dc, name="Degree centrality")

# Plot nodes centrality by changing some default parameters
plot_nodes_folium(G, attr_name='Degree centrality', tiles='openstreetmap', palette='viridis', 
                              zoom=14, fit_bounds=False, add_legend=True)
```
![image](https://user-images.githubusercontent.com/44501803/155641918-9f94916c-194c-45d1-a680-896b6d41e561.png)

For this option there is a [_make_folium_circlemarker](https://github.com/PyMap/ssyt/blob/main/ssyt/accesibilidad.py#L42-L107) method that is called/nested at a higher [plot_nodes_folium](https://github.com/PyMap/ssyt/blob/main/ssyt/accesibilidad.py#L7-L40) level where the graph structure is transformed into a Points Geodataframe data structure.

Here there are some arbitrary decisions like using the branca `to_step` method with 10 bins to go from linear to continuous/categorical intervals. Another valid option here would be to pick lower/higher colors from the [node_colors](https://github.com/PyMap/ssyt/blob/main/ssyt/accesibilidad.py#L35) and just show lower/higher level ticks. Like this:

![image](https://user-images.githubusercontent.com/44501803/155642762-ad27d941-f666-484e-8751-4141069e2490.png)

Beside this we could also include some additional parameter to also plot edges. Maybe inside the higher `plot_nodes_folium` method I referenced before by using a boolean `add_polyline`:

```
 gdf_nodes = ox.graph_to_gdfs(G)[0]

 if attr_name in gdf_nodes.columns:
    node_colors = ox.plot.get_node_colors_by_attr(G, attr_name, cmap=palette)
    gdf_nodes['nc'] = node_colors
else:
    raise ValueError('To plot {}, this must be added as a graph nodes attribute first'.format(attr_name))

# this returns a  folium.Map object
nodes_map = _make_folium_circlemarker(gdf_nodes, tiles, zoom, fit_bounds, attr_name, add_legend)

if add_polyline:
# add each edge to the map
    for vals in gdf[attrs].values:
        params = dict(zip(["geom", "popup_val"], vals))
        pl = _make_folium_polyline(**params, **kwargs)
        pl.add_to(nodes_map) 

```

And then decide if we want to show plot and edges together or just one by one with a:
```
folium.LayerControl('topleft', collapsed=True).add_to(m)
``` 
 
**2. Including the `plot_nodes_folium` inside the already existing `plot_graph_folium` method**. This, by using an extra parameter that gives the chance to plot nodes colored by attribute, maybe like this:

```
# create gdf of all graph edges
gdf_edges = utils_graph.graph_to_gdfs(G, nodes=False)
edges_map = _plot_folium(gdf_edges, graph_map, popup_attribute, tiles, zoom, fit_bounds, **kwargs)

if nodes_attr:
   nodes_map = _make_folium_circlemarker(gdf_nodes, tiles, zoom, fit_bounds, attr_name, add_legend)
   nodes_map.add_to(edges_map)
```

We can also plot nodes without coloring them, but I do not think this would be pretty usefulsince it doesn't add any useful information to the graph.

**Comment by gboeing** [comment]
This will be resolved by #957.
